lib/Ui/Tree.elm

```diff
@@ -0,0 +1,178 @@
+module Ui.Tree exposing (..)
+
+import Css
+import Extra.Tree
+import Html.Styled as Styled
+import Html.Styled.Attributes exposing (css)
+import Html.Styled.Events as StyledEvents
+import Tree exposing (Tree(..), label, tree)
+import Tree.Zipper as Zipper exposing (Zipper)
+import Ui.Color exposing (toCssColor)
+import Ui.Icon as Icon
+import Ui.Palette
+
+
+type alias Model a =
+    Zipper (Node a)
+
+
+init : Tree a -> Model a
+init =
+    Zipper.fromTree
+        << Tree.restructure identity
+            (\n c ->
+                case c of
+                    [] ->
+                        tree (Leaf n) []
+
+                    children ->
+                        tree (Collapsed n) children
+            )
+
+
+type Node a
```

Comment:
I had imagined that we only could see one row of children, together with a "back" button, rather than working with it like a collapsable directory tree

Reply:
Lets talk about this over zoom later!